src/index.ts

```diff
@@ -22,46 +23,35 @@ export default class TikiClient {
 
   /**
    * Initialize the TikiClient and register the device's address.
-   * @param {string} providerId - the provider ID of the associated provider account.
-   * @param {string} pubKey - the public key of the created provider.
    * @param {string} userId - the ID to be registered to identify the user.
    */
-  public static async initialize(
-    providerId: string,
-    pubKey: string,
-    userId: string
-  ): Promise<void> {
+  public static async initialize(userId: string): Promise<void> {
+    TikiClient.userId = userId;
 
-    TikiClient.userId = userId
+    const key = await TikiClient.keyService.get(TikiClient.config.providerId, TikiClient.userId);
 
-    const keys = await TikiClient.keyService.get();
-
-    if (keys.find((key) => key.value.name === `${providerId}.${userId}`))
+    if (
+     !key
+    )
       throw new Error(
```

Comment:
Why throw an error when the key does not exist?
I think that the correct would be to skip resgiterAddress if the key exists.